core/eolearn/tests/test_utils/test_testing.py

```diff
@@ -0,0 +1,140 @@
+            FeatureType.SCALAR_TIMELESS: "scalar_timeless",
+            FeatureType.LABEL_TIMELESS: "label_timeless",
+        },
+    ],
+)
+def test_generate_eopatch_seed(seed: int, features: FeaturesSpecification) -> None:
+    patch1 = generate_eopatch(features, seed=seed)
+    patch2 = generate_eopatch(features, seed=seed)
+    assert patch1 == patch2
+
+
+@dataclass
+class GenerateTestCase:
+    data_features: FeaturesSpecification
+    seed: int
+    expected_statistics: Dict[FeatureSpec, Dict[str, Any]]
+
+
+@pytest.mark.parametrize(
+    "test_case",
+    [
+        GenerateTestCase(
+            seed=3,
+            data_features=(FeatureType.MASK, "CLM"),
+            expected_statistics={
+                (FeatureType.MASK, "CLM"): {
+                    "exp_shape": (5, 98, 151, 2),
+                    "exp_min": 0,
+                    "exp_max": 255,
+                    "exp_mean": 127.38907960535208,
+                    "rel_delta": 0.0001,
+                }
+            },
+        ),
+        GenerateTestCase(
+            seed=1,
+            data_features={FeatureType.DATA: ["data"], FeatureType.MASK_TIMELESS: ["LULC", "IS_VALUE"]},
+            expected_statistics={
+                (FeatureType.DATA, "data"): {
+                    "exp_shape": (5, 98, 151, 1),
+                    "exp_min": -4.030404248695601,
+                    "exp_max": 4.406353522522504,
+                    "exp_mean": -0.00515678644681796,
+                    "rel_delta": 0.0001,
+                },
+                (FeatureType.MASK_TIMELESS, "LULC"): {
+                    "exp_shape": (98, 151, 2),
+                    "exp_min": 0,
+                    "exp_max": 255,
+                    "exp_mean": 127.03858629544533,
+                    "rel_delta": 0.0001,
+                },
+                (FeatureType.MASK_TIMELESS, "IS_VALUE"): {
+                    "exp_shape": (98, 151, 2),
+                    "exp_min": 0,
+                    "exp_max": 255,
+                    "exp_mean": 127.31409649952697,
+                    "rel_delta": 0.0001,
+                },
+            },
+        ),
+    ],
+)
+def test_generate_eopatch_data(test_case: GenerateTestCase) -> None:
+    patch = generate_eopatch(test_case.data_features, seed=test_case.seed)
+    for feature in parse_features(test_case.data_features):
+        assert_statistics_match(patch[feature], **test_case.expected_statistics[feature])
+
+
+@pytest.mark.parametrize(
+    "config",
+    [
+        {"num_timestamps": 3, "max_integer_value": 1, "raster_shape": (1, 1), "depth_range": (1, 2)},
+        {"num_timestamps": 7, "max_integer_value": 333, "raster_shape": (3, 27), "depth_range": (5, 15)},
+    ],
+)
+def test_generate_eopatch_config(config: Dict[str, Any]) -> None:
+    mask_feature = (FeatureType.MASK, "mask")
+
+    patch = generate_eopatch(mask_feature, config=PatchGeneratorConfig(**config))
+
+    time, raster_x, raster_y, depth = patch[mask_feature].shape
```

Comment:
AFAIK it's `time, height, width, depth`, so you got x and y mixed up (doesn't affect the test, just the names are switched)

Reply:
switch the names